packages/side-nav/src/SideNavItem.tsx

```diff
@@ -180,28 +164,36 @@ export interface SideNavItemProps {
  @param props.as When provided, the component will be rendered as the component or html tag indicated
  *  by this prop. Other additional props will be spread on the anchor element.
  */
-
 const SideNavItem: ExtendableBox<
   SideNavItemProps & { ref?: React.Ref<any> },
   'button'
 > = React.forwardRef((props: SideNavItemProps, forwardRef) => {
   const {
-    active = false,
+    active: activeProp,
     disabled = false,
     ariaCurrentValue = AriaCurrentValue.Page,
     className,
     children,
+    onClick: onClickProp,
+    glyph,
+    ...rest
   } = props;
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-  const rest = omit(props, [
-    'active',
-    'disabled',
-    'ariaCurrentValue',
-    'className',
-    'children',
-  ]);
+  const { currentPath } = useSideNavContext();
+  const active = activeProp != null || currentPath === rest.href;
```

Comment:
Do we need to support bot currentPath matching rest.href and active? I feel like sometimes we try and cast a wider net in favor of attempting to improve the DX, but sometimes this backfires on us. I'm not highly opinionated on this one, but wanted to call that out as something we should think about

Reply:
If the consumer uses `as` to render these as React router links with `to`, or something else, an `active` prop provides alternative means to set an element as active.

Reply:
Since the `currentPath` API is new, I suppose I could see removing it.

Reply:
Talked offline, and removed the API